Tests/UnitTests/Purchasing/Purchases/TransactionPosterTests.swift

```diff
@@ -48,8 +49,27 @@ class TransactionPosterTests: TestCase {
                 source: .init(isRestore: false, initiationSource: .queue)
             )
         )
-        expect(result).to(beFailure())
         expect(result.error) == BackendError.missingReceiptFile(self.receiptFetcher.mockReceiptURL)
+        expect(result.error?.finishable) == false
+
+        expect(self.cache.postedTransactions).to(beEmpty())
```

Comment:
`testHandlePurchasedTransactionWithMissingReceipt` is basically testing a not-finishable error. In this case we don't save the posted transaction.